File: store/model/profile_config.go
package model

import "time"

// HotspotProfileConfig menyimpan konfigurasi mikhmon-specific per profile hotspot.
// Digunakan expiry service untuk menentukan aksi saat user expired.
type HotspotProfileConfig struct {
	ID          uint           `gorm:"primaryKey"`
	DeviceID    uint           `gorm:"uniqueIndex:uniq_dev_prof;not null"`
	Device      MikrotikDevice `gorm:"foreignKey:DeviceID"`
	ProfileName string         `gorm:"uniqueIndex:uniq_dev_prof;not null;size:64"`

	// Mode expired: 0 (none) | rem (remove) | ntf (notice) | remc (remove+record) | ntfc (notice+record)
	ExpiryMode string `gorm:"default:'0';size:8"`
	Validity   string `gorm:"size:16"` // "30d", "1d", dll
	Price      int
	SellPrice  int
	LockMAC    bool `gorm:"default:false"`

	CreatedAt time.Time
	UpdatedAt time.Time
}

File: website/src/types/profile-config.ts
// Sesuai dto.ProfileConfigResponse backend (snake_case 1:1).
//
// expiry_mode enum:
//   - "0"   = none / disable (free profile, tidak ada selling record)
//   - "rem" = hapus user saat expired
//   - "ntf" = set limit-uptime ke 0 + kick (notice)
//   - "remc"= hapus + catat transaksi
//   - "ntfc"= notice + catat transaksi
export type ExpiryMode = '0' | 'rem' | 'ntf' | 'remc' | 'ntfc'

export interface ProfileConfig {
  id: number
  device_id: number
  profile_name: string
  expiry_mode: ExpiryMode
  validity: string
  price: number
  sell_price: number
  lock_mac: boolean
  created_at: string
  updated_at: string
}

// Body untuk PUT /devices/{id}/hotspot/profile-configs/{profile_name}.
export interface ProfileConfigUpsertRequest {
  expiry_mode: ExpiryMode
  validity: string
  price: number
  sell_price: number
  lock_mac: boolean
}

// Sesuai dto.ProfileConfigSyncResponse backend.
export interface ProfileConfigSyncResponse {
  synced: string[]
  created: string[]
  orphan: string[]
  injected: string[]
  inject_failed: string[]
}
